scripts/gen-did/main.ts

```diff
@@ -89,88 +78,91 @@ function exportKeypairs(keypairs: {
             pubKey: u8aToHex(keypairs.keyAgreement.publicKey),
             seed: keypairs.baseMnemonic + '//did//keyAgreement//0',
             privKey: u8aToHex(keypairs.keyAgreement.secretKey),
-        }
+        },
     }
 }
 
 async function writeDidSecretsFile(didSecrets: DidSecrets, filename: string) {
-    await fs.writeFile(filename, JSON.stringify(didSecrets, null, 2));
+    await fs.writeFile(filename, JSON.stringify(didSecrets, null, 2))
 }
 
-
 async function main() {
     // connect to kilt node
-    let endpoint = 'wss://spiritnet.kilt.io/';
+
+    let endpoint = 'wss://peregrine.kilt.io/'
     if (process.env.ENDPOINT) {
         switch (process.env.ENDPOINT) {
-            case 'spiritnet': endpoint = 'wss://spiritnet.kilt.io/'; break;
-            case 'peregrine': endpoint = 'wss://peregrine.kilt.io/'; break;
-            default: endpoint = process.env.ENDPOINT;
+            case 'spiritnet':
+                endpoint = 'wss://spiritnet.kilt.io/'
+                break
+            case 'peregrine':
+                endpoint = 'wss://peregrine.kilt.io/'
+                break
+            default:
+                endpoint = process.env.ENDPOINT
         }
     }
-    console.debug(`Connecting to ${endpoint}...`);
-    const api = await Kilt.connect(endpoint);
-    console.debug(`Connected to ${endpoint}`);
+    console.debug(`Connecting to ${endpoint}...`)
 
+    const api = await Kilt.connect(endpoint)
+    console.debug(`Connected to ${endpoint}`)
+    // console.log('api', api.on)
     // get first command line argument as seed for the payment account and fail if not provided
-    const paymentSeed = process.argv[2];
+    const paymentSeed = process.argv[2]
+    console.debug(paymentSeed)
     if (!paymentSeed) {
         // console.error('Please provide a seed for the payment account as first command line argument.');
-        process.exit(1);
+        console.error('Payment seed not found')
+        process.exit(1)
     }
-    let paymentKeyPair = Kilt.Utils.Crypto.makeKeypairFromSeed(
-        mnemonicToMiniSecret(paymentSeed),
-        'sr25519',
-    );
-    console.debug(`Payment account address: ${paymentKeyPair.address}`);
+    const didSeed = process.argv[2]
+
+    const paymentKeyPair = Kilt.Utils.Crypto.makeKeypairFromUri(paymentSeed, 'sr25519')
```

Comment:
since we use 'sr25519' multiple times here we maybe should put it into a constant, so that you can change the key type by changing a single constant.

That would of course only make sense if all the key types are for the same key. If you need to configure each keytype separately, a single constant for all wouldn't be such a great idea.

Reply:
Added this into the utils file